ibis_substrait/compiler/translate.py

```diff
@@ -1449,6 +1454,9 @@ def _check_and_upcast(op: ops.Node) -> ops.Node:
     """Check that arguments to extension functions have consistent types."""
     op_name = IBIS_SUBSTRAIT_OP_MAPPING[type(op).__name__]
 
+    if any(isinstance(arg, ops.TableArrayView) for arg in op.args):
+        raise NotImplementedError("Subqueries are unsupported as function inputs")
+
```

Comment:
Would ask you to double check my thinking here.

Before 7.1, `tpc_h17` was failing because `_rewrite_filter` wasn't implemented for `Aggregation`. 

With 7.1 [(this change)](https://github.com/ibis-project/ibis/pull/7368/), we make it past the rewrite filter stuff but `tpc_h17` still fails [because of this assertion](https://github.com/ibis-project/ibis-substrait/blob/4e8f2eb034a99c865b2e57fb8f786c750e9e893f/ibis_substrait/compiler/translate.py#L811-L813). It seems that something isn't right with selections that rely on subqueries.

We get similar failures on other TPC query tests with subqueries (H2 and H15). And I gather from the `xfail` annotations that Substrait translation isn't entirely implemented for subqueries and continues to not work with 7.1+, i.e. it's not something we need to fix _for_ 7.1.

I added this assertion to explicitly "unsupport" subquery translation (instead of implicitly failing on [that assertion](https://github.com/ibis-project/ibis-substrait/blob/4e8f2eb034a99c865b2e57fb8f786c750e9e893f/ibis_substrait/compiler/translate.py#L811-L813) as we do for other queries).

This obviously has the risk of failing something that wasn't failing before. Happy to undo and `xfail` H17 on an `AssertionError` same as H2 and H15.

Reply:
I think that your analysis here is correct and this is a reasonable solution.  We do need to do a more thorough job with subqueries, but we're currently rewriting a bunch of Ibis internals (ongoing) so I'm good with leaving this in an explicitly unsupported state for now.